examples/nodejs/src/index.ts

```diff
@@ -0,0 +1,125 @@
+import * as ScanbotSDK from "scanbotsdk";
+
+import { printUsage } from "./snippets/utils/example-usage";
+import { parseFlags } from "./snippets/utils/utils";
+import { DetectBarcodesSnippet } from "./snippets/barcode/detect-barcodes";
+import { DetectDocumentSnippet } from "./snippets/document/detect-document";
+import { CheckScannerSnippet } from "./snippets/datacapture/check-scanner";
+import { CreditCardScannerSnippet } from "./snippets/datacapture/credit-card-scanner";
+import { DocumentDataExtractorSnippet } from "./snippets/datacapture/document-data-extractor";
+import { MedicalCertificateScannerSnippet } from "./snippets/datacapture/medical-certificate-scanner";
+import { MrzScannerSnippet } from "./snippets/datacapture/mrz-scanner";
+import { TextPatternScannerSnippet } from "./snippets/datacapture/text-pattern-scanner";
+import { OcrSnippet } from "./snippets/datacapture/ocr";
+import { VinScannerSnippet } from "./snippets/datacapture/vin-scanner";
+import { DocumentClassifierSnippet } from "./snippets/document/document-classifier";
+import { MrzParserSnippet } from "./snippets/datacapture/mrz-parser";
+import { ParseBarcodeDocumentSnippet } from "./snippets/barcode/parse-barcode-document";
+import { AnalyzeMultiPageSnippet } from "./snippets/document/analyse-multipage";
+import { CropAndAnalyzeSnippet } from "./snippets/document/crop-analyse";
+
+async function awaitPromise(promise: Promise<void>, maxAwaitTimeMs: number = 60 * 1000): Promise<void> {
+  const timer = setTimeout(() => {
+    throw new Error("Operation timed out");
+  }, maxAwaitTimeMs);
+  await promise;
+  clearTimeout(timer);
+}
+
+async function main(): Promise<void> {
+  if (process.argv.length < 2) {
+    printUsage();
+    return;
+  }
+
+  const args = process.argv.slice(2);
+
+  const category = args[0]?.toLowerCase() ?? "";
+  const subcommand = args[1]?.toLowerCase() ?? "";
+  const flags = parseFlags(args.slice(2));
+
+  const file = flags["--file"] as string | undefined;
+  const save = flags["--save"] as string | undefined;
+  const text = flags["--text"] as string | undefined;
+  const licenseFlag = flags["--license"] as string | undefined;
+
+  const licenseKey: string = licenseFlag ?? "SCANBOTSDK-LICENSE";
+  const writablePath = ".";
+
+  await ScanbotSDK.initialize({
+    licenseKey,
+    writablePath,
+  });
+
+  const licenseInfo = await ScanbotSDK.getLicenseInfo();
+  console.log("License Status:", licenseInfo.status);
+
+  const isFloatingLicense = !!licenseInfo.devices;
+  if (isFloatingLicense) {
+    console.log(
+      `Using floating license with ${licenseInfo.devices} devices. Do not forget to call deregisterDevice ` +
+        `when you no longer need the license.`
+    );
+  }
+
+  await ScanbotSDK.autorelease(async () => {
```

Comment:
Let's add comment why the autorelease is used and that alternatively (or in addition) we can write `await using image = ...`